opensearch/src/main/java/org/opensearch/sql/opensearch/setting/OpenSearchSettings.java

```diff
@@ -149,6 +151,27 @@ public class OpenSearchSettings extends Settings {
           Setting.Property.NodeScope,
           Setting.Property.Dynamic);
 
+  public static final Setting<TimeValue> SESSION_INDEX_TTL_SETTING =
+      Setting.positiveTimeSetting(
+          Key.SESSION_INDEX_TTL.getKeyValue(),
+          timeValueDays(14),
+          Setting.Property.NodeScope,
+          Setting.Property.Dynamic);
+
+  public static final Setting<TimeValue> RESULT_INDEX_TTL_SETTING =
+      Setting.positiveTimeSetting(
+          Key.RESULT_INDEX_TTL.getKeyValue(),
+          timeValueDays(60),
+          Setting.Property.NodeScope,
+          Setting.Property.Dynamic);
+
+  public static final Setting<Boolean> AUTO_INDEX_MANAGEMENT_ENABLED_SETTING =
+      Setting.boolSetting(
+          Key.AUTO_INDEX_MANAGEMENT_ENABLED.getKeyValue(),
+          true,
+          Setting.Property.NodeScope,
+          Setting.Property.Dynamic);
+
   /** Construct OpenSearchSetting. The OpenSearchSetting must be singleton. */
   @SuppressWarnings("unchecked")
   public OpenSearchSettings(ClusterSettings clusterSettings) {
```

Comment:
Probably using this object can avoid creating new UpdateConsumers in ClusterManagerEventListeners on index settings. This class will always have the latest values.